src/fsharp/ParseAndCheckInputs.fs

```diff
@@ -686,11 +686,21 @@ let TypeCheckOneInputEventually (checkForErrors, tcConfig: TcConfig, tcImports:
               let conditionalDefines =
                   if tcConfig.noConditionalErasure then None else Some (tcConfig.conditionalCompilationDefines)
 
+              let hadSig = rootSigOpt.IsSome
+
               // Typecheck the implementation file 
-              let! topAttrs, implFile, _implFileHiddenType, tcEnvAtEnd, createsGeneratedProvidedTypes = 
-                  TypeCheckOneImplFile (tcGlobals, tcState.tcsNiceNameGen, amap, tcState.tcsCcu, checkForErrors, conditionalDefines, tcSink, tcConfig.internalTestSpanStackReferring) tcImplEnv rootSigOpt file
+              let typeCheckOne = 
+                  if skipImplIfSigExists && hadSig then
```

Comment:
Just out of curiosity, is hadsig the right name?  It seems as if it should be hasSig.

Reply:
`hadSig` refers to if the impl file "had" a sig before we type-checked.  `hasSig` might mean if the impl has any kind of signature, whether it be generated after a type-check or from a '.fsi' file.

Honestly, `hasSig` might also be correct depending on how you look at it.

The most accurate name would be `hasSigFromSigFile/hadSigFromSigFile`.